WalletWasabi.Fluent/ViewModels/Wallets/BtcAmount.cs

```diff
@@ -0,0 +1,24 @@
+using System.Reactive.Linq;
+using NBitcoin;
+using WalletWasabi.Fluent.Infrastructure;
+
+namespace WalletWasabi.Fluent.ViewModels.Wallets;
+
+public class BtcAmount
+{
+	private BtcAmount(Money value, IExchangeRateProvider exchangeRateProvider)
+	{
+		Value = value;
+		UsdValue = exchangeRateProvider.BtcToUsdRate.Select(x => x * Value.ToDecimal(MoneyUnit.BTC));
+		ExchangeRates = exchangeRateProvider.BtcToUsdRate;
+	}
+
+	public Money Value { get; }
+	public IObservable<decimal> UsdValue { get; }
+	public IObservable<decimal> ExchangeRates { get; }
+
+	public static BtcAmount Create(Money value)
+	{
+		return new BtcAmount(value, new ExchangeRateProvider(Services.Synchronizer));
```

Comment:
Doesn't the static reference here go against the UI Decoupling?
Every viewmodel that crates a `BtcAmount` object will implicitly reference to `Services.Synchronizer`

Reply:
Yes, you're absolutely right. This is going exactly the opposite 😑

@ichthus1604 I'd like to know your opinion on this. We're creating amounts all along, and they need the exchange rate. Is there a handy yet elegant solution to keep VMs decoupled and testable, while not creating much hundle?